composite/tree/task.rb

```diff
@@ -8,9 +8,7 @@ def time_required
     if leaf?
       @time_required
     else
-      time = 0.0
-      children.each { |child_task| time += child_task.time_required }
-      time
+      children.inject(0.0) { |time, task| time += child_task.time_required }
```

Comment:
I think there is a typo and it should be or `task` in the equation or `child_task` in block argument.